src/main/java/com/microsoft/azure/proton/transport/proxy/ProxyConfiguration.java

```diff
@@ -0,0 +1,119 @@
+package com.microsoft.azure.proton.transport.proxy;
+
+import com.microsoft.azure.proton.transport.proxy.impl.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.PasswordAuthentication;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class ProxyConfiguration implements AutoCloseable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProxyConfiguration.class);
+
+    private final String proxyAddress;
```

Comment:
This should be a java.net.Proxy instead of a string. Other SDKs use that class for the address+port so we should too.